secure-api-gateway-ob-uk-rs-server/src/main/java/com/forgerock/sapi/gateway/ob/uk/rs/server/api/obie/payment/v3_1_5/domesticstandingorders/DomesticStandingOrdersApiController.java

```diff
@@ -120,9 +118,10 @@ public ResponseEntity<OBWriteDomesticStandingOrderResponse6> createDomesticStand
         JsonObject intent = consentService.getIDMIntent(authorization, consentId);
         log.debug("Retrieved consent from IDM");
 
-        //deserialize the intent to ob response object
-        OBWriteDomesticStandingOrderConsentResponse6 consent = consentService.deserialize(
-                OBWriteDomesticStandingOrderConsentResponse6.class,
+        // deserialize the intent to ob response object where:
+        // The initiation object class from the consent must be the same class of the payment request initiation object
+        OBWriteDomesticStandingOrderConsentResponse5 consent = consentService.deserialize(
+                OBWriteDomesticStandingOrderConsentResponse5.class,
```

Comment:
Can you explain this change please?

Are the initiation types not the same in the Consent and Standing Order payment request for the same API version?

Reply:
Just that, the Standing order payment submission initiation section is not the same type that the initiation object from the consent, so that we deserialise the consent response object from OBIntentObject stored in IDM to a type where the initiation object is the same type of payment submission request to apply the .equals method properly.

This deserialise is used only to build the consent object to validate it, It is not used to build any API responses. 

The API respond with the proper expected type defined by the spec.

We could deserialise only the initiation instead to deserialise all consent to apply the initiation validation, but we need validate also the Risk section.

Reply:
Ok, I think this can be cleaned up when we move consents to mongo